persistence/provider/memory/transaction.go

```diff
@@ -18,27 +18,31 @@ type transaction struct {
 }
 
 // Commit applies the changes from the transaction.
-func (t *transaction) Commit(ctx context.Context) error {
+func (t *transaction) Commit(
+	ctx context.Context,
+) (*persistence.TransactionResult, error) {
 	defer t.end()
 
 	if t.ds == nil {
-		return persistence.ErrTransactionClosed
+		return nil, persistence.ErrTransactionClosed
 	}
 
 	if err := t.ds.checkOpen(); err != nil {
-		return err
+		return nil, err
 	}
 
 	if !t.hasLock {
-		return nil
+		return nil, nil
```

Comment:
```suggestion
		return &persistence.TransactionResult{}, nil
```

This is a transaction in which nothing happened, so it still needs a valid result, it's just empty. This means we're missing a test-case that an empty `TransactionResult` is returned when no operations are done in the transaction. That should go in `providertest/transaction.go`.

Reply:
I would assume that this change will be required not only for in-memory engine, right?

Reply:
I think it's only this one, but just add the test and you'll know :D